generators/getMetric.js

```diff
@@ -0,0 +1,86 @@
+const path = require('path')
+const fs = require('fs')
+const { metricsList } = require('./getMetricFunctions')
+const fsExtra = require('fs-extra')
+
+function generateGetMetric () {
+  let generatedDoc = ''
+  for (const metric of metricsList) {
+    generatedDoc += allTimeBoundTemplates(metric)
+  }
+  let file = path.join(`${__dirname}`, `../src/getMetric.js`)
+  if (fs.existsSync(file)) {
+    fsExtra.removeSync(file)
+  }
+  return fs.writeFileSync(file, generatedDoc)
+}
+
+function allTimeBoundTemplates (metric) {
+  if (!('currency' in metric)) {
+    metric.currency = ''
+  }
+  if (!('sheetMetricName' in metric)) {
+    metric.sheetMetricName = metric.metric
+  }
+  if (!('timeBound' in metric)) {
+    metric.timeBound = ''
+  }
+  let generatedString =
+    fillTemplate(metric.metric, metric.sheetMetricName, metric.description, metric.currency, metric.timeBound)
+
+  return generatedString
+}
+
+function fillTemplate (metric, sheetMetricName, description, currency, timeBound) {
+  let [options, bonusDescription, functionArguments] =
+    optionsGenerator(currency, timeBound)
+  return `/**
+* Gets ${description}
+* @param {string} projectSlug Name of the asset at sanbase,
+* which can be found at the end of the URL (eg. the Santiment URL is
+* https://app.santiment.net/projects/santiment, so the projectSlug would be santiment).
+* @param {date} from The starting date to fetch the data. Example: DATE(2018, 9, 20)
+* @param {date} to The ending date to fetch the data. Example: DATE(2018, 9, 21)${bonusDescription}
+* @returns {number} returns ${description}.
+* @customfunction
+*/
+function SAN_${sheetMetricName.toUpperCase()} (${functionArguments}) {
+  return handleErrors_(getMetric_)('${metric}', projectSlug, from, to, { ${options} })
+}
+
+`
+}
+
+function optionsGenerator (currency, timeBound) {
+  let returnedValues = {
+    description: [],
+    arguments: ['projectSlug', 'from', 'to'],
+    options: []
+  }
+  if (currency === true) {
```

Comment:
let's rename to `hasCurrency` or something similar. Just `currency == true` sounds to me like the project is part of this list: https://api.santiment.net/graphiql?query=%7B%0A%20%20allCurrencyProjects%20%7B%0A%20%20%20%20slug%0A%20%20%7D%0A%7D%0A

Reply:
You're right, just updated it, also I made it so, that I do not switch the variable's type (if (!(currency in metric)) currency = ''), now it is renamed and it is = false

Reply:
https://github.com/santiment/google-spreadsheet-addon/pull/67/files#diff-a1047c598842ea2deec85d4f81629e25R28